webhook_server/tests/test_log_viewer.py

```
@@ -0,0 +1,1217 @@
+            mock_config_instance.data_dir = str(tmp_path)
+            mock_config.return_value = mock_config_instance
+            return LogViewerController(logger=mock_logger)
+
+    async def test_shutdown_closes_websocket_connections(self, controller):
+        """Test shutdown closes all WebSocket connections."""
+        # Create mock WebSocket connections
+        ws1 = Mock()
+        ws2 = Mock()
+
+        # Make close() async
+        async def mock_close(code, reason):
```

Ruff (0.14.10)

Missing return type annotation for private function `mock_close`

Add return type annotation: `None`

(ANN202)